warehouse/query-core/src/main/java/datawave/query/iterator/logic/TermFrequencyExcerptIterator.java

```diff
@@ -302,9 +344,38 @@ public void next() throws IOException {
             // get the next term frequency
             source.next();
         }
-
         // generate the return key and value
-        tk = new Key(row, new Text(dtUid), new Text(fieldName + Constants.NULL + generatePhrase(terms)), cv, ts);
+        String phraseWithScores = generatePhrase(terms);
+        boolean usedScores = false;
+        // check to see if we have something scored and if so, turn off outputting the scores
+        for (WordsAndScores term : terms) {
+            if (term == null) {
+                continue;
+            }
+            if (term.getUseScores()) {
+                usedScores = true;
+                term.setOutputScores(false);
+            }
+        }
```

Comment:
Could we avoid calling `generatePhase` twice if we check this first or even check the boolean value of `useScores` here?

Reply:
done